core/trino-spi/src/main/java/io/trino/spi/type/Int128Math.java

```diff
@@ -31,7 +31,7 @@
     private static final int NUMBER_OF_LONGS = 2;
     private static final int NUMBER_OF_INTS = 2 * NUMBER_OF_LONGS;
 
-    private static final Int128[] POWERS_OF_TEN = new Int128[38]; // 10^38 is the largest value < Int128.MAX_VALUE
+    private static final Int128[] POWERS_OF_TEN = new Int128[39]; // 10^38 is the largest value < Int128.MAX_VALUE
```

Comment:
update comment on why we have 39 here

Reply:
No, the comment is still valid. The previous array bound was incorrect, as it caused the array to contain only powers 10^0 to 10^37.